week2/NextJs.md

```diff
@@ -0,0 +1,187 @@
+# Next.js 🙂
+
+Created: Jun 18, 2021 10:02 AM
+
+# Next.js
+
+Next.js는 2016년에 Vercel 이라는 곳에서 처음 만든 **React 프론트엔드 웹 프레임워크**입니다.
+
+## 왜 Next.js 을 쓸까?
+
+### 이미지 최적화
+
+이미지 컴포넌트 `next/image` 를 만들었습니다. 모던 웹에 잘 맞습니다.
+
+**이미지 리사이즈, 최적화, `WebP`와 같은 모던 포맷을 제공**합니다. 이것이 이미지 자동 최적화입니다.
+
+- WebP 는 PNG, JPEG 보다 컬러의 깊이감, 애니메이션 프레임, 불투명도 등에서 더 나은 압축 방식입니다. 크롬, 엣지, 파이어폭스, 오페라, 사파리에서 지원해요. ~~인터넷익스플로러 안녕~~
+- 화면의 표시 영역(viewport)이 작다면 그에 맞춰 굳이 큰 이미지를 보이지 않도록 합니다.
+- 외부에서 이미지를 가져오더라도 다 동작 합니다.
+
+`Webpack` 의 경우 이미지 최적화는 빌드 시에 수행되는데, `Next.js` 에서는 유저가 **요청**할 때, 즉 수요에 맞추어 최적화 합니다. 따라서 일반적인 정적 사이트를 만드는 툴과 다르게 `Next.js` 의 빌드 시간은 이미지 개수에 영향을 받지 않습니다.
+
+이미지는 디폴트로 **lazy loading** 됩니다. 스크롤해서 viewport에 들어와 있는 이미지들이 로드됩니다.
```

Comment:
https://css-tricks.com/the-complete-guide-to-lazy-loading-images/
https://developer.mozilla.org/en-US/docs/Web/API/Intersection_Observer_API

I have no idea about how Nextjs implements image lazy loading internally but the links above might be interesting. Thanks!